### PR TITLE
chore(mux-uploader): move file-select to own component

### PR DESCRIPTION
**Pull request by davekiss**
- breaks out file-selector to its own component
- add tests
- breaking: renames the slot from `upload-button` to `file-select`

### CODE REVIEW

File: packages/mux-uploader/src/mux-uploader-file-select.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  handleFilePickerButtonClick() {
+    // TO-DO: Allow user to reattempt uploading the same file after an error.
+    // Note: Apparently Chrome and Firefox do not allow changing an indexed property on FileList...(TD).
+    // Source: https://stackoverflow.com/a/46689013
+    const attr = this.getAttribute('mux-uploader');
+    const controller = attr ? document.getElementById(attr) : (<ShadowRoot>this.getRootNode()).host;
+
+    controller?.shadowRoot?.querySelector<HTMLInputElement>('#hidden-file-input')?.click();
```
Comment (cjpillsbury) [inline packages/mux-uploader/src/mux-uploader-file-select.ts:92]: question(blocking): Shouldn't the `#hidden-file-input` be owned by this component as well? Seems like an appropriate separation of concerns/single responsibility here. You'd then just have it dispatch the same event as `MuxUploaderDropElement` and those "look identical" from the perspective of the uploader element.

File: packages/mux-uploader/src/mux-uploader-file-select.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+}
+</style>
+
+<slot>
+<button type="button">Upload video</button>
+</slot>
+`;
+
+class MuxUploaderFileSelectElement extends globalThis.HTMLElement {
+  protected _filePickerButton: HTMLElement | null | undefined;
```
Comment (cjpillsbury) [inline packages/mux-uploader/src/mux-uploader-file-select.ts:47]: nitpick(blocking): We've been moving to using native javascript `#private` variables. Change for consistency.

File: packages/mux-uploader/src/mux-uploader-file-select.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+      this.filePickerButton = slot.assignedElements({ flatten: true })[0] as HTMLButtonElement;
+    });
+  }
+
+  protected get filePickerButton() {
+    return this._filePickerButton;
+  }
+
+  protected set filePickerButton(value: HTMLElement | null | undefined) {
```
Comment (cjpillsbury) [inline packages/mux-uploader/src/mux-uploader-file-select.ts:73]: praise: smart to use the setter here.

File: packages/mux-uploader/src/mux-uploader-file-select.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    if (value === this._filePickerButton) return;
+    if (this._filePickerButton) {
+      this._filePickerButton.removeEventListener('click', this.handleFilePickerButtonClick);
+    }
+
+    this._filePickerButton = value;
+    if (this._filePickerButton) {
+      this._filePickerButton.addEventListener('click', this.handleFilePickerButtonClick);
```
Comment (cjpillsbury) [inline packages/mux-uploader/src/mux-uploader-file-select.ts:81]: thought(non-blocking): Could we make this `useCapture` to easily handle even more cases? I could see some negatives there on assumptions with that approach, so definitely more "food for thought" right now.

**Review by cjpillsbury — CHANGES_REQUESTED** [review]
A couple of callouts, but overall I'm loving the direction this is going.

File: packages/mux-uploader/src/mux-uploader-file-select.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    }
+  }
+
+  handleFilePickerButtonClick() {
+    // TO-DO: Allow user to reattempt uploading the same file after an error.
+    // Note: Apparently Chrome and Firefox do not allow changing an indexed property on FileList...(TD).
+    // Source: https://stackoverflow.com/a/46689013
+    const attr = this.getAttribute('mux-uploader');
+    const controller = attr ? document.getElementById(attr) : (<ShadowRoot>this.getRootNode()).host;
```
Comment (cjpillsbury) [inline packages/mux-uploader/src/mux-uploader-file-select.ts:90]: suggestion(non-blocking): I believe you can use `(this.getRootNode() as ShadowRoot).host` here.

File: packages/mux-uploader/src/mux-uploader-file-select.ts (hunk 1 above)
Comment (davekiss) [reply packages/mux-uploader/src/mux-uploader-file-select.ts:92]: yes, I think so too, but i need to get the event bus in place first for that to work which is in another branch at the moment. can we address this in another PR?

Comment (cjpillsbury) [reply packages/mux-uploader/src/mux-uploader-file-select.ts:92]: Deal.

**Review by cjpillsbury — APPROVED** [review]
🚀

**Pull request merged**